### DISCUSSION TITLE
[Start] Is there a way to access the window object during beforeLoad in Tanstack Start like you can in Tanstack Router?

### DISCUSSION THREAD

**Discussion by guilhermohounie** [TanStack/router#3034] [Q&A]
I'm migrating an app from tanstack router to start because I will need some server side functionalities, but what I was doing before is not working when migrating.

For example, when using tanstack router only:

```ts
export const Route = createFileRoute("/dashboard")({
  component: RouteComponent,
  pendingComponent: LoadingPage,
  head: () => ({ meta: [{ title: "Dashboard" }] }),
  beforeLoad: ({ context }) => {
  // here, poketbase reads from the localStorage for the auth key, and it works correctly because the window object is defined here in tanstack router
     console.log(window) // works!
    const { pocketbase } = context;
    if (!pocketbase.authStore.isValid) {
      throw redirect({ to: "/login" });
    }
  }
});
```

However, trying to do the exact same with tanstack start results in an error 

```ts
export const Route = createFileRoute("/dashboard")({
  component: RouteComponent,
  pendingComponent: LoadingPage,
  head: () => ({ meta: [{ title: "Dashboard" }] }),
  beforeLoad: ({ context }) => {
     console.log(window) // window is not defined, therefore an error is thrown
     / ....
  }
});
```

Can I tell tanstack start "hey, this has to be rendered completely at the client side, so the beforeLoad will have the window context in it" ?

**Comment by schiller-manuel** [comment]
no, `beforeLoad` will always run on the server when SSRing.

I have never used pocketbase, it has a section about SSR though: https://www.npmjs.com/package/pocketbase#ssr-integration

So if you can find a way to create an implementation for the server, you can create an isomorphic function via `createIsormophicFn` and call that in beforeLoad.

If you want to discuss details, head over to our discord https://tlinz.com/discord
